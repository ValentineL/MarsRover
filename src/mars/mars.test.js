import React from 'react';
import { shallow } from 'enzyme';
import Mars from './mars';

describe('Mars', () => {
    const props = {
        position: "0 0 N",
        execute: true,
        commands: 'MMRMMLMMRM',
        size: 5,
        startPosition: '00N'
    };

    const initialState = {
        start: null,
        end: null,
        ops: [],
        position: "0 0 N",
        facing: "N",
        path: null,
        error: null
    };

    it('should render Mars component', () => {
        const component = shallow(<Mars {...props} />);
        expect(component).toMatchSnapshot();
    });
    it('should render Mars component with initial state', () => {
        let component = shallow(<Mars {...props} />);
        component.setState({ state: initialState }, () => { component.update(); });
    });
});