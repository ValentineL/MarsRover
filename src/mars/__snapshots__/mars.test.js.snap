// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mars should render Mars component 1`] = `
<ul
  className="mars"
>
  <li
    className="cell  "
    key="0-4"
  >
    <label>
      0-4
    </label>
  </li>
  <li
    className="cell  "
    key="1-4"
  >
    <label>
      1-4
    </label>
  </li>
  <li
    className="cell  "
    key="2-4"
  >
    <label>
      2-4
    </label>
  </li>
  <li
    className="cell  "
    key="3-4"
  >
    <label>
      3-4
    </label>
  </li>
  <li
    className="cell  "
    key="4-4"
  >
    <label>
      4-4
    </label>
  </li>
  <li
    className="cell  "
    key="0-3"
  >
    <label>
      0-3
    </label>
  </li>
  <li
    className="cell  "
    key="1-3"
  >
    <label>
      1-3
    </label>
  </li>
  <li
    className="cell  "
    key="2-3"
  >
    <label>
      2-3
    </label>
  </li>
  <li
    className="cell  "
    key="3-3"
  >
    <label>
      3-3
    </label>
  </li>
  <li
    className="cell  "
    key="4-3"
  >
    <label>
      4-3
    </label>
  </li>
  <li
    className="cell  "
    key="0-2"
  >
    <label>
      0-2
    </label>
  </li>
  <li
    className="cell  "
    key="1-2"
  >
    <label>
      1-2
    </label>
  </li>
  <li
    className="cell  "
    key="2-2"
  >
    <label>
      2-2
    </label>
  </li>
  <li
    className="cell  "
    key="3-2"
  >
    <label>
      3-2
    </label>
  </li>
  <li
    className="cell  "
    key="4-2"
  >
    <label>
      4-2
    </label>
  </li>
  <li
    className="cell  "
    key="0-1"
  >
    <label>
      0-1
    </label>
    <Rover
      facing="N"
    />
  </li>
  <li
    className="cell  "
    key="1-1"
  >
    <label>
      1-1
    </label>
  </li>
  <li
    className="cell  "
    key="2-1"
  >
    <label>
      2-1
    </label>
  </li>
  <li
    className="cell  "
    key="3-1"
  >
    <label>
      3-1
    </label>
  </li>
  <li
    className="cell  "
    key="4-1"
  >
    <label>
      4-1
    </label>
  </li>
  <li
    className="cell path  start"
    key="0-0"
  >
    <label>
      0-0
    </label>
    <Rover
      facing="N"
      ghost={true}
    />
  </li>
  <li
    className="cell  "
    key="1-0"
  >
    <label>
      1-0
    </label>
  </li>
  <li
    className="cell  "
    key="2-0"
  >
    <label>
      2-0
    </label>
  </li>
  <li
    className="cell  "
    key="3-0"
  >
    <label>
      3-0
    </label>
  </li>
  <li
    className="cell  "
    key="4-0"
  >
    <label>
      4-0
    </label>
  </li>
</ul>
`;
